client.go

```diff
@@ -258,19 +313,19 @@ func (z *Client) SafeGet(path string) ([]byte, *zk.Stat, error) {
 	return z.Get(path)
 }
 
-// fullpath returns the path with the chroot prepended. It can cause issues
-// if the znode path starts like /foo/foo/...
+// fullpath returns the path with the chroot prepended.
 func (z *Client) fullpath(path string) string {
-	if z.chroot != "" && !strings.HasPrefix(path, z.chroot) {
-		return z.chroot + path
-	}
+	//	if z.Cfg.Chroot != "" && !strings.HasPrefix(path, z.Cfg.Chroot) {
+	//    return z.Cfg.Chroot + path
+	//	}
+	//	return path
 
-	return path
+	return z.Cfg.Chroot + path
```

Comment:
This can cause issues with watchers, the zk.Event.Path will contain the chroot, if this is used to do some operation we will be appending the chroot twice.